game_patch/multi/server.cpp

```diff
@@ -305,6 +316,36 @@ static void send_private_message_with_stats(rf::Player* player)
     send_chat_line_packet(str.c_str(), player);
 }
 
+void shuffle_level_array()
+{
+    std::size_t n = rf::netgame.levels.size();
+    for (std::size_t i = n - 1; i > 0; --i) {
+        std::size_t j = dist(rng) % (i + 1);
+        std::swap(rf::netgame.levels[i], rf::netgame.levels[j]);
+    }
```

Comment:
Consider `std::shuffle`. It can take `rng` as a parameter.

Reply:
I did not realize `std::shuffle` existed - that is much more streamlined. Thank you!